snippets/csharp/System.Net.Http/HttpCompletionOption/HttpCompletionOptionSnippets.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace HttpCompletionOptionSnippets
+{
+    class HttpCompletionOptionSnippets
+    {
+        public static async Task RunAsync()
+        {
+            //<SnippetHttpCompletionOption>
+            var httpClient = new HttpClient();
+            httpClient.Timeout = TimeSpan.FromSeconds(30);
+            using (var response = await httpClient.GetAsync("http://localhost:12345/", HttpCompletionOption.ResponseHeadersRead)) // 30-second timeout but ONLY up until past the headers
+            {
+                // Do other stuff that doesn't rely on the content first, like status code validation
+                response.EnsureSuccessStatusCode();
+
+                var content = await response.Content.ReadAsStringAsync(); // NO TIMEOUT

```

Comment:
The solution is different for .NET Framework vs .NET. In .NET, ReadAsStringAsync overrides accept a CancellationToken: https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpcontent.readasstringasync?view=net-9.0#system-net-http-httpcontent-readasstringasync(system-threading-cancellationtoken). In .NET Framework, this override doesn't exist, and so this workaround is usually used by developers: https://stackoverflow.com/questions/25985416/how-can-i-set-a-timeout-for-an-async-function-that-doesnt-accept-a-cancellation.

Adding this to the example would mean either:
a) Adding code with conditional compilation (`#if NETFRAMEWORK /*timeout workaround*/ #else /*cancellation token version*/ #endif`)
b) Adding separate examples for .NET vs .NET Framework

I don't know if this take away clarity from the point we're illustrating with the example though, and I don't know what the community's preference would be for such case.